x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/create_index_route.ts

```diff
@@ -94,3 +145,62 @@ export const createDetectionIndex = async (
     await createBootstrapIndex(esClient, index);
   }
 };
+
+const addFieldAliasesToIndices = async ({
+  esClient,
+  index,
+  spaceId,
+}: {
+  esClient: ElasticsearchClient;
+  index: string;
+  spaceId: string;
+}) => {
+  const { body: indexMappings } = await esClient.indices.get({ index });
+  // Make sure that all signal fields we add aliases for are guaranteed to exist in the mapping for ALL historical
+  // signals indices (either by adding them to signalExtraFields or ensuring they exist in the original signals
+  // mapping) or else this call will fail and not update ANY signals indices
+  const fieldAliases = createSignalsFieldAliases();
+  for (const [indexName, mapping] of Object.entries(indexMappings)) {
+    const currentVersion: number | undefined = get(mapping.mappings?._meta, 'version');
+    const newMapping = {
+      properties: {
+        ...signalExtraFields,
+        ...fieldAliases,
+        ...getRbacRequiredFields(spaceId),
+      },
+      _meta: {
+        version: currentVersion,
+        [ALIAS_VERSION_FIELD]: SIGNALS_FIELD_ALIASES_VERSION,
+      },
+    };
+    await esClient.indices.putMapping({
+      index: indexName,
+      body: newMapping,
+      allow_no_indices: true,
+    } as estypes.IndicesPutMappingRequest);
```

Comment:
Seems like it sequentially updates mappings of every concrete index matching the alias (`index`). If there's a lot of concrete indices, this can take additional time spent for network roundtrips etc.

Is there a reason why we can't update all concrete indices at once via the alias?

If yes, should we maybe split it into buckets of N indices (e.g. 10 or 25), run updates in parallel within each bucket and process buckets sequentially?

Reply:
Yeah bucketing these requests and making them in parallel would be much better. Unfortunately we can't do them all at once because we need to extract the original `version` from the mapping to preserve it, but we also need to add the `aliases_version` so that we know which indices have had the compatibility-aliases added. If we do it all at once, then if we apply the mapping across indices with different versions then the versions will be wrong for at least some of them.

Reply:
> Unfortunately we can't do them all at once because we need to extract the original `version` from the mapping to preserve it

Is it because the PUT mappings request replaces the whole `_meta` object? Sounds counter-intuitive since PUT mappings behaves more like a PATCH, so I'd expect it to replace `_meta.aliases_version` and preserve `_meta.version`.

Maybe I'm just not getting this (if yes, sorry about that):

> If we do it all at once, then if we apply the mapping across indices with different versions then the versions will be wrong for at least some of them.

Reply:
Oh yeah PUT mapping behaves like a normal PUT with the `_meta` field unlike the mappings and replaces the whole thing. I was expecting the same thing you did initially and tried including only the `aliases_version` in the meta object and it ended up removing the `version` field.